### PR TITLE
Adds Docker Compose to ease local development - Resolves #85

### LINKED ISSUE #85: Add docker-compose to run entire env in local
Currently is not that easy to setup the dev environment as you need to create your own supabase db and manually add a lot of envs to the apps/web/.env file.

The main goal of this issue is to fulfill the happy path by locally spinning up all the required containers for the app.

### Happy path
- Run `docker-compose up`  on the root
- Starts NextJS + Supabase with proper environment pointing to the supabase local db
- Go to `localhost:3000`
- Insert email
- Click on the CTA
- See the confetti (email is added to the local DB)

### PR DESCRIPTION
**Pull request by jppradoleal**
This pull request adds Supabase Docker Compose specs, and creates a base Compose spec that includes it and defines a service for `apps/web` with a volume mirroring it (to keep the hot reload feature).

It also defines a Makefile to ease the process of creating the .env files and running the migrations, I've splitted the db:migrate as `generate-types` seemed unnecessary.

---

I've used the email-sender Dockerfile as base, which already follows the best conventions, but the image size is kind of a pain, with almost 4gb, the main cause is the dependencies. `yarn install` layer is the biggest one, I dont think we can optimize it without pruning some dependencies.

image

Considering that the image must be rebuilt only when the dependencies changes, it's not a problem as big as it seems.

---

Some things I did forget to mention:

1. It doesn't include `auto-email-sender-compose.yml`, as it seems optional.
2. I had to bump up the supabase-cli version, to use the --db-url attribute, as it was unsupported in previous versions.

### CODE REVIEW

**Comment by albertofp** [comment]
Let's fucking goooo
Tiny nit: maybe add more to the .dockerignore to reduce image size a bit more (dockerfile, logs, readmes etc)
Not sure how the --prune flag works but maybe .dockerignore-ing **/node-modules helps?

**Comment by jppradoleal** [comment]
The .dockerignore is actually a symbolic link to the .gitignore, I am not sure if Docker is able to parse it as it should. I will try to hardcode some values in it, and see if the size changes.

**Comment by jppradoleal** [comment]
Yeah, `.dockerignore` as a symbolic link to `.gitignore` is parsed as it should. Hardcoding the values from one into another didn't change the image size.

File: docker/.env.example
Comment (ocodista) [inline docker/.env.example:1]: It seems to me that this entire `docker` folder is actually a `supabase` image folder.

What do you think about renaming?

File: apps/web/.dockerignore
Comment (ocodista) [inline apps/web/.dockerignore:1]: Does this works?

We need to make sure that no .git, folder is being pushed to the docker image.

**Review by ocodista — CHANGES_REQUESTED** [review]

File: apps/web/.dockerignore
Comment (jppradoleal) [reply apps/web/.dockerignore:1]: image

Yeah, it does work :)

File: docker/.env.example
Comment (jppradoleal) [reply docker/.env.example:1]: Done 👍

**Comment by jppradoleal** [comment]
I've renamed the docker folder to supabase, and did some cleaning on its folders.

**Pull request closed without merging**